mountpoint-s3-client/src/object_client.rs

```diff
@@ -266,6 +266,13 @@ pub enum GetObjectAttributesError {
     NoSuchKey,
 }
 
+/// Type of Server side encryption(SSE) opted for Put Object
+#[derive(Debug, Clone)]
+pub enum KmsKeys {
+    SseKms,
+    DsseKms,
+}
```

Comment:
This isn't KMS type but rather Server-side encryption (SSE) type.

We might include SSE-S3, SSE-C, or something new in the future.

Reply:
Each of these variants will potentially have a different set of config associated with it. So maybe make this a richer enum type:
```rust
pub enum ServerSideEncryption {
    /// Use the default configuration for the bucket
    #[default]
    Default,
    Kms {
        key_id: Option<String>,
    },
    DualLayerKms {
        key_id: Option<String>,
    },
    // not now, but in the future
    CustomerProvided {
        key: Key,
    }
}
```